flang/lib/Lower/IO.cpp

```diff
@@ -462,8 +462,8 @@ genOutputItemList(Fortran::lower::AbstractConverter &converter,
       if (fir::isa_complex(itemTy)) {
         auto parts = fir::factory::ComplexExprHelper{builder, loc}.extractParts(
             itemValue);
-        outputFuncArgs.push_back(parts.first);
-        outputFuncArgs.push_back(parts.second);
+        outputFuncArgs.push_back(builder.createConvert(loc, argType, parts.first));
+        outputFuncArgs.push_back(builder.createConvert(loc, argType, parts.second));
```

Comment:
Ideally, shouldn't this be calling a function to output an `f128`?

I think I'd prefer to call a fictitious output function and have a linker error than silently truncate the value. We do so elsewhere when the runtime doesn't support all precisions.